modules/internal/String.chpl

```diff
@@ -430,31 +433,62 @@ module String {
       }
     }
 
+    /*
+      Iterates over the string Unicode character by Unicode character.
+    */
+    iter uchars(): int(32) {
+      var localThis: string = this.localize();
+
+      var i = 0;
+      while i < localThis.len {
+        var codepoint: int(32);
+        var nbytes: c_int;
+        var multibytes = (localThis.buff + i): c_string;
+        var maxbytes = (localThis.len - i): ssize_t;
+        if maxbytes > 4 then
```

Comment:
Does it help anything to have this conditional here? I would have expected qio_decode_char_buf to handle counts > 4. Does it not?

Reply:
Right, checking the code in qio_decode_char_buf, it does handle the larger counts, so I will delete those tests.